Data_Structures_Answers.md

```diff
@@ -2,17 +2,39 @@ Add your answers to the questions below.
 
 1. What is the runtime complexity of your ring buffer's `append` method?
 
+O(1).
+
 2. What is the space complexity of your ring buffer's `append` function?
 
+O(1). Unless you are counting the bits allocated through appending itself, which you might be, in which case I suppose it would be technically O(n), but within such a tiny range of n, and you've probably already allocated it, that who cares? There is no real additional space cost to store the reference.
+
 3. What is the runtime complexity of your ring buffer's `get` method?
 
+O(n) because it temporarily allocates a new list to use for purging None entries.
+
 4. What is the space complexity of your ring buffer's `get` method?
 
+O(n) because it temporarily allocates a new list to use for purging None entries. This only affects RAM though, so I would not even count it when discussing disk space complexity, as it were.
+
 
 5. What is the runtime complexity of the provided code in `names.py`?
 
+O(n * n') because it iterates over the entire second list for every instance in the first list. Though it's kind of weird to call it n^2, I believe it's correct nonetheless, because n implies it's a single variable, whereas there are "two" n's here.
+
+Nonetheless that's why the "second part" of n is designated by n'.
+
 6. What is the space complexity of the provided code in `names.py`?
 
+O(n), because it just reads out two lists.
```

Comment:
Consider that this just iterates through 2 lists without creating anything new in memory.  Because of this we can consider it an O(1) solution in terms of space complexity.  However, the argument for O(n) still exists if are considering the duplicates list as extra allocated memory for this method.